sdks/rust/golem-rust-macro/src/agentic/agent_definition_impl.rs

```diff
@@ -0,0 +1,195 @@
+                    }
+                }
+            }
+
+            let name = &trait_fn.sig.ident;
+            let method_name = &name.to_string();
+
+            let mut description = String::new();
+
+            for attr in &trait_fn.attrs {
+                if attr.path().is_ident("description") {
+                    let mut found = None;
+                    attr.parse_nested_meta(|meta| {
+                        if meta.path.is_ident("description") {
+                            let lit: syn::LitStr = meta.value()?.parse()?;
+                            found = Some(lit.value());
+                            Ok(())
+                        } else {
+                            Err(meta.error("expected `description = \"...\"`"))
+                        }
+                    })
+                    .ok();
+                    if let Some(val) = found {
+                        description = val;
+                    }
+                }
+            }
+
+
+            let mut parameter_types = vec![]; // This is WIT type for now, but needs to support structured text type
+            let mut result_type = vec![];
+
+            if let syn::TraitItem::Fn(trait_fn) = item {
+                for input in &trait_fn.sig.inputs {
+                    if let syn::FnArg::Typed(pat_type) = input {
+                        let param_name = match &*pat_type.pat {
+                            syn::Pat::Ident(pat_ident) => pat_ident.ident.to_string(),
+                            _ => "_".to_string(), // fallback for patterns like destructuring
+                        };
+                        let ty = &pat_type.ty;
+                        parameter_types.push(quote! {
+                            (#param_name.to_string(), golem_rust::golem_agentic::golem::agent::common::ElementSchema::ComponentModel(<#ty as ::golem_rust::agentic::Schema>::get_wit_type()))
+                        });
+                    }
+                }
+
+                // Handle return type
+                match &trait_fn.sig.output {
+                    syn::ReturnType::Default => (),
+                    syn::ReturnType::Type(_, ty) => {
+                        result_type.push(quote! {
+                            ("return-value".to_string(),   golem_rust::golem_agentic::golem::agent::common::ElementSchema::ComponentModel(<#ty as ::golem_rust::agentic::Schema>::get_wit_type()))
+                        });
+                    }
+                };
+            }
+
+            let input_parameters = parameter_types;
+            let output_parameters = result_type;
+
+
+            Some(quote! {
+                ::golem_rust::golem_agentic::golem::agent::common::AgentMethod {
+                    name: #method_name.to_string(),
+                    description: #description.to_string(),
+                    prompt_hint: None,
+                    input_schema: ::golem_rust::golem_agentic::golem::agent::common::DataSchema::Tuple(vec![#(#input_parameters),*]),
```

Comment:
Both input and output are hardcoded here to be `Tuple`, no concept for how it's going to support multimodal